src/dune/lib_rules.ml

```diff
@@ -209,12 +211,11 @@ let build_shared lib ~sctx ~dir ~flags ~is_empty =
             ; Dep src
             ]
         in
-          if ctx.ccomp_type = "msvc" && is_empty then
-            cmd
-          else
-            Build.dyn_paths_unit
-              (Build.return [ Path.build (Library.archive lib ~dir ~ext:ext_lib) ])
-            >>> cmd
+        if ctx.ccomp_type = "msvc" && is_empty then
+          cmd
+        else
+          Build.path (Path.build (Library.archive lib ~dir ~ext:ext_lib))
```

Comment:
The change from `Build.dyn_paths_unit (Build.return [Path.build …])` has affected the test results (see Travis)

Reply:
Ah indeed.  The new output looks fine, but I'll do this in another PR